subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -1096,34 +1111,41 @@ public Object configure(Object object, Closure configureClosure) {
         return objects;
     }
 
+    private <T> Iterable<T> configure(Iterable<T> objects, Closure configureClosure, String callerContextInformation) {
+        for (T object : objects) {
+            gradle.getDomainObjectConfigurator().configure(object, configureClosure, callerContextInformation);
+        }
+        return objects;
+    }
+
     @Override
     public void configurations(Closure configureClosure) {
         ((Configurable<?>) getConfigurations()).configure(configureClosure);
     }
 
     @Override
     public void repositories(Closure configureClosure) {
-        ConfigureUtil.configure(configureClosure, getRepositories());
+        gradle.getDomainObjectConfigurator().configure(getRepositories(), configureClosure, null);
```

Comment:
There are only a few cross-configuration callbacks that we want to track with build operations, the rest does not need to be wrapped like this.